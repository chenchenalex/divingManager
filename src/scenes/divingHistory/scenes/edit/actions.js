import { ADD_DIVE, EDIT_DIVE } from "src/actions/actionTypes";

export const addNewDive = formData => ({
  type: ADD_DIVE,
  payload: formData
});

export const editDive = formData => ({
  type: EDIT_DIVE,
  payload: formData
});
