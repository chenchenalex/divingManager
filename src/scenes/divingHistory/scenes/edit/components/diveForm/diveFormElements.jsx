import React from "react";
import TextField from "@material-ui/core/TextField";
import PropTypes from "prop-types";
import { DivingForm } from "../../style";
import LocationAutosuggest from "./locationSuggestion";

const formElements = ({
  classes,
  handleChange,
  handleBlur,
  formData,
  formConfig,
  locationSuggestions,
  onSuggestionsFetchRequested,
  onSuggestionsClearRequested,
  getSuggestionValue,
  inputProps
}) => (
  <DivingForm>
    <TextField
      fullWidth
      className={classes && classes.formControl}
      id="name"
      label="Name"
      error={formConfig.name.invalid}
      helperText={formConfig.name.invalid && formConfig.name.helperText}
      required={formConfig.name.required}
      value={formData && formData.name}
      onBlur={handleBlur("name")}
      onChange={handleChange("name")}
      margin="normal"
    />

    <LocationAutosuggest
      suggestions={locationSuggestions}
      onSuggestionsFetchRequested={onSuggestionsFetchRequested}
      onSuggestionsClearRequested={onSuggestionsClearRequested}
      getSuggestionValue={getSuggestionValue}
      inputProps={inputProps}
    />

    <TextField
      fullWidth
      id="date"
      helperText={formConfig.date.invalid && formConfig.date.helperText}
      required={formConfig.date.required}
      type="date"
      error={formConfig.date.invalid}
      value={formData && formData.date}
      label={formConfig.date.label}
      onBlur={handleBlur("date")}
      onChange={handleChange("date")}
      margin="normal"
    />

    <TextField
      fullWidth
      id="diveBuddy"
      helperText={
        formConfig.diveBuddy.invalid && formConfig.diveBuddy.helperText
      }
      required={formConfig.diveBuddy.required}
      type="text"
      error={formConfig.diveBuddy.invalid}
      value={formData && formData.diveBuddy}
      label={formConfig.diveBuddy.label}
      onBlur={handleBlur("diveBuddy")}
      onChange={handleChange("diveBuddy")}
      margin="normal"
    />

    <TextField
      select
      fullWidth
      id="rating"
      helperText={formConfig.rating.invalid && formConfig.rating.helperText}
      required={formConfig.rating.required}
      type="text"
      error={formConfig.rating.invalid}
      value={formData && formData.rating}
      label={formConfig.rating.label}
      SelectProps={{ native: true }}
      onBlur={handleBlur("rating")}
      onChange={handleChange("rating")}
      margin="normal"
    >
      {formConfig.rating.options.map(i => (
        <option key={i} value={i}>
          {i}
        </option>
      ))}
    </TextField>

    <TextField
      fullWidth
      id="depth"
      helperText={formConfig.depth.invalid && formConfig.depth.helperText}
      required={formConfig.depth.required}
      type="number"
      value={formData && formData.depth}
      error={formConfig.depth.invalid}
      label={formConfig.depth.label}
      onBlur={handleBlur("depth")}
      onChange={handleChange("depth")}
      margin="normal"
    />

    <TextField
      fullWidth
      id="oxygen"
      helperText={formConfig.oxygen.invalid && formConfig.oxygen.helperText}
      required={formConfig.oxygen.required}
      type="number"
      value={formData && formData.oxygen}
      error={formConfig.oxygen.invalid}
      label={formConfig.oxygen.label}
      onBlur={handleBlur("oxygen")}
      onChange={handleChange("oxygen")}
      margin="normal"
    />

    <TextField
      fullWidth
      id="feedback"
      helperText={formConfig.feedback.invalid && formConfig.feedback.helperText}
      required={formConfig.feedback.required}
      multiline={formConfig.feedback.multiline}
      rows={formConfig.feedback.rowsMax}
      value={formData && formData.feedback}
      error={formConfig.feedback.invalid}
      label={formConfig.feedback.label}
      onBlur={handleBlur("feedback")}
      onChange={handleChange("feedback")}
      margin="normal"
    />
  </DivingForm>
);

formElements.propTypes = {
  classes: PropTypes.object,
  handleChange: PropTypes.func.isRequired,
  handleBlur: PropTypes.func.isRequired,
  formData: PropTypes.object,
  formConfig: PropTypes.object.isRequired,
  locationSuggestions: PropTypes.array.isRequired,
  onSuggestionsFetchRequested: PropTypes.func.isRequired,
  onSuggestionsClearRequested: PropTypes.func.isRequired,
  getSuggestionValue: PropTypes.func.isRequired,
  inputProps: PropTypes.object.isRequired
};

formElements.defaultProps = {
  classes: null,
  formData: null
};

export default formElements;
