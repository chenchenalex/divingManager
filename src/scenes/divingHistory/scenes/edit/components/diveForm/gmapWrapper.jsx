import React from "react";
import {
  withGoogleMap,
  withScriptjs,
  GoogleMap,
  Marker
} from "react-google-maps";
import PropTypes from "prop-types";
import { compose, withProps } from "recompose";
import credentials from "src/services/apiCredentials";

const GMapWrapper = compose(
  withProps({
    googleMapURL: `
    https://maps.googleapis.com/maps/api/js?key=${
      credentials.gmap
    }&v=3.exp&libraries=geometry,drawing,places`,
    loadingElement: <div style={{ height: "100%" }} />,
    containerElement: (
      <div style={{ height: "500px", width: "500px", margin: "46px" }} />
    ),
    mapElement: <div style={{ height: "100%" }} />
  }),
  withScriptjs,
  withGoogleMap
)(({ markerPos, onDragEnd }) => (
  <GoogleMap
    defaultZoom={8}
    disableDefaultUI
    defaultCenter={markerPos || { lat: -34.397, lng: 150.644 }}
  >
    {markerPos && (
      <Marker draggable clickable position={markerPos} onDragEnd={onDragEnd} />
    )}
  </GoogleMap>
));

GMapWrapper.propTypes = {
  markerPos: PropTypes.object,
  onDragEnd: PropTypes.func.isRequired
};

GMapWrapper.defaultProps = {
  markerPos: null
};

export default GMapWrapper;
