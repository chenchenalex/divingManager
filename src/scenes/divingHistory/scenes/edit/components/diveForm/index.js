import diveFrom from "./diveForm";

export default diveFrom;
