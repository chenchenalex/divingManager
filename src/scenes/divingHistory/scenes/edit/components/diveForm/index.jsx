import React from "react";
import Loader from "src/components/loader";
import Loadable from "react-loadable";
import PropTypes from "prop-types";

const LoadableComponent = Loadable({
  loader: () => import("./diveForm"),
  loading: Loader
});

const asyncForm = ({ history, match }) => (
  <LoadableComponent history={history} match={match} />
);

asyncForm.propTypes = {
  history: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired
};

export default asyncForm;
