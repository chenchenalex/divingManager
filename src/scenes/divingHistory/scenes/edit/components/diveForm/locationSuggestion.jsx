import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Autosuggest from "react-autosuggest";
import TextField from "@material-ui/core/TextField";
import Paper from "@material-ui/core/Paper";
import MenuItem from "@material-ui/core/MenuItem";
import match from "autosuggest-highlight/match";
import parse from "autosuggest-highlight/parse";

const style = theme => ({
  container: {
    flexGrow: 1,
    position: "relative"
  },
  suggestionsContainerOpen: {
    position: "absolute",
    zIndex: 1,
    marginTop: theme.spacing.unit,
    left: 0,
    right: 0
  },
  suggestion: {
    display: "block"
  },
  suggestionsList: {
    margin: 0,
    padding: 0,
    listStyleType: "none"
  }
});

function renderSuggestion(suggestion, { query, isHighlighted }) {
  const matches = match(suggestion.name, query);
  const parts = parse(suggestion.name, matches);

  return (
    <MenuItem selected={isHighlighted} component="div">
      <div>
        {parts.map(
          (part, index) =>
            part.highlight ? (
              <span key={String(index)} style={{ fontWeight: 300 }}>
                {part.text}
              </span>
            ) : (
              <strong key={String(index)} style={{ fontWeight: 500 }}>
                {part.text}
              </strong>
            )
        )}
      </div>
    </MenuItem>
  );
}

function renderSuggestionsContainer(options) {
  const { containerProps, children } = options;

  return (
    <Paper {...containerProps} square>
      {children}
    </Paper>
  );
}

function renderInput(inputProps) {
  const {
    ref,
    required,
    onBlur,
    helperText,
    invalid,
    label,
    ...other
  } = inputProps;

  return (
    <TextField
      fullWidth
      required={required}
      onBlur={onBlur}
      error={invalid}
      helperText={invalid && helperText}
      label={label}
      margin="normal"
      InputProps={{
        inputRef: ref,
        ...other
      }}
    />
  );
}

const locationSuggestion = ({
  classes,
  suggestions,
  onSuggestionsFetchRequested,
  onSuggestionsClearRequested,
  getSuggestionValue,
  inputProps
}) => (
  <Autosuggest
    theme={{
      container: classes.container,
      suggestionsContainerOpen: classes.suggestionsContainerOpen,
      suggestionsList: classes.suggestionsList,
      suggestion: classes.suggestion
    }}
    renderInputComponent={renderInput}
    suggestions={suggestions}
    renderSuggestionsContainer={renderSuggestionsContainer}
    onSuggestionsFetchRequested={onSuggestionsFetchRequested}
    onSuggestionsClearRequested={onSuggestionsClearRequested}
    getSuggestionValue={getSuggestionValue}
    renderSuggestion={renderSuggestion}
    inputProps={inputProps}
  />
);

locationSuggestion.propTypes = {
  classes: PropTypes.object.isRequired,
  suggestions: PropTypes.array.isRequired,
  onSuggestionsFetchRequested: PropTypes.func.isRequired,
  onSuggestionsClearRequested: PropTypes.func.isRequired,
  getSuggestionValue: PropTypes.func.isRequired,
  inputProps: PropTypes.object.isRequired
};

export default withStyles(style)(locationSuggestion);
