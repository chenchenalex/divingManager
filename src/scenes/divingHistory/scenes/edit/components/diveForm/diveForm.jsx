import React from "react";
import Button from "material-ui/Button";
import { connect } from "react-redux";
import store from "src/store";
import Loader from "src/components/loader";

import { Link } from "react-router-dom";
import DiveFormElements from "./diveFormElements";
import { INITIAL_FORM_DATA, FORM_CONFIG } from "src/data/config";
import { LOCATION_LIST } from "src/data/mockData";
import { FormActions } from "../../style";
import { userFetchDataAsync } from "../../../../actions";

// import action
import { addNewDive, editDive } from "../../actions";

const classes = {
  container: "divingform-container",
  formControl: "form-control",
  label: "form-label",
  input: "text-input",
  button: "form-button"
};

const { dispatch } = store;

export class DivingForm extends React.Component {
  formConfig = JSON.parse(JSON.stringify(FORM_CONFIG)); // deep copy config

  componentDidMount() {
    if (!this.props.match.params.id) return;

    if (this.state.editingFormData) {
      this.validateForm();
    } else {
      // Dive requested not found, request database
      dispatch(userFetchDataAsync("alex"));
      // this.props.history.push("/404");
    }

    console.log("component did mount");
  }

  static getDerivedStateFromProps(props) {
    if (Object.keys(props.divingHistory.diveById).length > 0) {
      // Edit existing Dive
      if (props.match.params.id) {
        const {
          divingHistory: { diveById }
        } = props;

        const savedForm = diveById[props.match.params.id];

        return {
          formData: savedForm,
          editingFormData: savedForm,
          locationSuggestions: [],
          isTouched: false
        };
      } else {
        return {
          formData: { ...INITIAL_FORM_DATA },
          editingFormData: { ...INITIAL_FORM_DATA },
          locationSuggestions: [],
          isTouched: false
        };
      }
    }
    return {};
  }

  initializeState = props => {
    if (props.match.params.id) {
      // Edit existing dive
      const {
        divingHistory: { diveById }
      } = props;

      this.savedForm = diveById[props.match.params.id];

      this.setState(state => {
        return {
          formData: this.savedForm,
          editingFormData: this.savedForm,
          locationSuggestions: [],
          isTouched: false
        };
      });
    } else {
      // Add new dive
      this.setState(state => {
        return {
          formData: { ...INITIAL_FORM_DATA },
          editingFormData: { ...INITIAL_FORM_DATA },
          locationSuggestions: [],
          isTouched: false
        };
      });
    }
  };

  handleChange = name => (event, { newValue } = {}) => {
    // This is how to deep clone an object
    const newVal = newValue || event.target.value || "";

    this.setState(function(prevState) {
      const newState = {
        ...prevState,
        editingFormData: {
          ...prevState.editingFormData,
          [name]: newVal
        },
        isTouched: true
      };

      return newState;
    });
  };

  handleBlur = name => () => {
    const val = this.state.editingFormData[name];
    this.formConfig[name].invalid = this.checkFormFieldInValidity(name, val);

    this.validateForm();
  };

  resetAllForms = () => {
    Object.keys(this.formConfig).forEach(fieldKey => {
      this.formConfig[fieldKey].invalid = false;
    });

    this.setState(function(prevState) {
      return {
        editingFormData: { ...prevState.formData },
        isTouched: false
      };
    }, this.validateForm);
  };

  onFormSubmit = () => {
    let actionCreator;

    if (!this.state.formValid) return;

    // Call actions based on form ID
    if (typeof this.state.editingFormData.id === "undefined") {
      actionCreator = addNewDive(this.state.editingFormData);
    } else {
      actionCreator = editDive(this.state.editingFormData);
    }

    store.dispatch(actionCreator);

    this.props.history.push("/list");
  };

  checkFormFieldInValidity = (fieldKey, val) => {
    // don't check values that's not configured
    if (!this.formConfig[fieldKey]) return true;

    // Check empty values
    if (this.formConfig[fieldKey].required && (val === "" || val === null)) {
      return true;
    }

    // Check specific rules
    switch (fieldKey) {
      case "depth":
        return parseFloat(val) < 0;
      default:
        return false;
    }
  };

  validateForm = () => {
    let formInvalid = Object.keys(this.formConfig).some(key => {
      const isRequired = this.formConfig[key].required;

      const isEmpty = this.state.editingFormData[key] === "";

      const isInvalid = this.formConfig[key].invalid;

      return isRequired && (isEmpty || isInvalid);
    });

    this.setState(prevState => {
      return {
        ...prevState,
        formValid: !formInvalid
      };
    });
  };

  getSuggestions = input => {
    const inputValue = input.trim().toLowerCase();
    const inputLength = inputValue.length;
    let count = 0;

    return inputLength === 0
      ? []
      : LOCATION_LIST.filter(suggestion => {
          const keep =
            count <= 5 &&
            suggestion.name.toLowerCase().slice(0, inputLength) === inputValue;

          if (keep) count += 1;
          return keep;
        });
  };

  onSuggestionsFetchRequested = ({ value }) => {
    this.setState({
      locationSuggestions: this.getSuggestions(value)
    });
  };

  onSuggestionsClearRequested = () => {
    this.setState({
      locationSuggestions: []
    });
  };

  getSuggestionValue = suggestion => {
    return suggestion.name;
  };

  getInputPropsFromConfig = ({ label, required, invalid, helperText }) => {
    return {
      placeholder: helperText,
      value: this.state.editingFormData && this.state.editingFormData.country,
      onChange: this.handleChange("country"),
      onBlur: this.handleBlur("country"),
      type: "search",
      invalid,
      helperText,
      label,
      required
    };
  };

  render() {
    const { locationSuggestions, editingFormData, isTouched } = this.state;

    const locationInputProps = this.getInputPropsFromConfig(
      this.formConfig.country
    );

    return this.state.editingFormData ? (
      <div className={classes.container}>
        <DiveFormElements
          classes={classes}
          locationSuggestions={locationSuggestions}
          onSuggestionsFetchRequested={this.onSuggestionsFetchRequested}
          onSuggestionsClearRequested={this.onSuggestionsClearRequested}
          getSuggestionValue={this.getSuggestionValue}
          inputProps={locationInputProps}
          handleChange={this.handleChange}
          handleBlur={this.handleBlur}
          formData={editingFormData}
          formConfig={this.formConfig}
        />

        <FormActions>
          <Button
            className={classes.button}
            variant="raised"
            disabled={!this.state.formValid}
            color="primary"
            onClick={this.onFormSubmit.bind(this)}
          >
            Save
          </Button>
          <Link to="/list">
            <Button variant="raised">Back</Button>
          </Link>
          <Button
            variant="raised"
            disabled={!isTouched}
            onClick={this.resetAllForms}
          >
            Reset
          </Button>
        </FormActions>
      </div>
    ) : (
      <Loader />
    );
  }
}

function mapStateToProps(state) {
  return {
    divingHistory: state.scenes.divingHistory
  };
}

export default connect(mapStateToProps)(DivingForm);
