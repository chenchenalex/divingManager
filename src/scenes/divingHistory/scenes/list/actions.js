import { DELETE_DIVE } from "src/actions/actionTypes";

const deleteActionCreator = (itemsToBeDeleted = []) => {
  return {
    type: DELETE_DIVE,
    data: itemsToBeDeleted
  };
};

export { deleteActionCreator };
