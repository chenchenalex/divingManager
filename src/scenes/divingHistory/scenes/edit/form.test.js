import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import DiveFormElements from "./components/diveForm/diveformElements";
import { INITIAL_FORM_DATA, FORM_CONFIG } from "src/data/config";
import {
  TEST_INITIAL_DATA,
  divingHistory as mockData
} from "src/data/mockData";
import * as formActions from "./actions";

import { DivingForm } from "./components/diveForm/diveForm";

describe("DiveForm: add new dive tests", () => {
  const expectedState = {
    formData: { ...INITIAL_FORM_DATA },
    editingFormData: { ...INITIAL_FORM_DATA }
  };

  const match = {
    params: {}
  };

  const component = shallow(<DivingForm match={match} />);

  it("onLoad: add new dive should see blank page with initial data", () => {
    const instance = component.instance();

    expect(instance.state).toEqual(expectedState);
  });
});

describe("DiveForm: edit existing dive tests", () => {
  const match = {
    params: {
      id: "test1"
    }
  };

  const {
    diveById: { test1 }
  } = mockData;

  const expectedState = {
    formData: test1,
    editingFormData: test1,
    formValid: true
  };

  const component = shallow(
    <DivingForm match={match} divingHistory={mockData} history={[]} />
  );

  it("onLoad: edit existing dive should populate data correctly", () => {
    const instance = component.instance();

    expect(instance.state).toEqual(expectedState);
  });

  it("handleChange: should update local state reflecting the change", () => {
    const instance = component.instance();

    const handleNameChange = instance.handleChange("name");
    const event = {
      target: {
        value: "atlantic"
      }
    };
    handleNameChange(event);
    expect(instance.state.editingFormData.name).toBe("atlantic");

    const handleDateChange = instance.handleChange("date");
    const event1 = {
      target: {
        value: "20190101"
      }
    };

    handleDateChange(event1);
    expect(instance.state.editingFormData.date).toBe("20190101");
  });

  it("resetAllForms: should reset form to initial state", () => {
    const instance = component.instance();

    // This will mutate object and affect other tests
    instance.formConfig.name.invalid = true;
    instance.formConfig.depth.invalid = true;

    instance.setState(prevState => {
      return {
        editingFormData: {
          id: "test1",
          name: "new name",
          location: "new place",
          date: "123123",
          depth: 28
        },
        formValid: false
      };
    });

    // magical happens here
    instance.resetAllForms();

    expect(instance.formConfig.name.invalid).toBe(false);
    expect(instance.formConfig.depth.invalid).toBe(false);
    expect(instance.state.editingFormData).toEqual(instance.state.formData);
    expect(instance.state.formValid).toBe(true);
  });

  it("onFormSubmit: should dispatch action contains edited form data", () => {
    const instance = component.instance();

    instance.setState(state => {
      return { editingFormData: {}, formValid: true };
    });

    // Add new dive form submit
    const mockAddDive = jest.spyOn(formActions, "addNewDive");

    instance.onFormSubmit();

    expect(mockAddDive).toBeCalledWith(instance.state.editingFormData);

    instance.setState(state => {
      return { editingFormData: { ...test1 }, formValid: true };
    });

    // Edit dive form submit
    const mockEditDive = jest.spyOn(formActions, "editDive");

    instance.onFormSubmit();

    expect(mockEditDive).toBeCalledWith(instance.state.editingFormData);
  });
});

describe("DiveForm snapshot tests", () => {
  it("Dive form elements should render correctly", () => {
    const onChange = jest.fn();
    const onBlur = jest.fn();
    const formConfig = { ...FORM_CONFIG };

    const wrapper = renderer.create(
      <DiveFormElements
        handleChange={onChange}
        handleBlur={onBlur}
        formData={TEST_INITIAL_DATA}
        formConfig={formConfig}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
