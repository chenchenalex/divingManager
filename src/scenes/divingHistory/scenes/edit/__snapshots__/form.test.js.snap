// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiveForm snapshot tests Dive form elements should render correctly 1`] = `
<form
  className="sc-bdVaJa eyQCFM"
>
  <div
    className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-10 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8"
      data-shrink={true}
      htmlFor="name"
    >
      Name
      <span
        className="MuiFormLabel-asterisk-14"
      >
         *
      </span>
    </label>
    <div
      className="MuiInput-root-15 MuiInput-fullWidth-22 MuiInput-formControl-16 MuiInput-underline-19"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        className="MuiInput-input-23"
        disabled={false}
        id="name"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value="VIRGIN Dive with AOW licence"
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-10 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8"
      data-shrink={true}
      htmlFor="location"
    >
      Location
      <span
        className="MuiFormLabel-asterisk-14"
      >
         *
      </span>
    </label>
    <div
      className="MuiInput-root-15 MuiInput-fullWidth-22 MuiInput-formControl-16 MuiInput-underline-19"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        className="MuiInput-input-23 MuiInput-inputType-26 MuiInput-inputTypeSearch-27"
        disabled={false}
        id="location"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="search"
        value="PHILIPINE"
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-10 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8"
      data-shrink={true}
      htmlFor="date"
    >
      Date
      <span
        className="MuiFormLabel-asterisk-14"
      >
         *
      </span>
    </label>
    <div
      className="MuiInput-root-15 MuiInput-fullWidth-22 MuiInput-formControl-16 MuiInput-underline-19"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        className="MuiInput-input-23 MuiInput-inputType-26"
        disabled={false}
        id="date"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="date"
        value="2018-03-15"
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-10 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-shrink-8"
      data-shrink={true}
      htmlFor="depth"
    >
      Depth
    </label>
    <div
      className="MuiInput-root-15 MuiInput-fullWidth-22 MuiInput-formControl-16 MuiInput-underline-19"
    >
      <input
        aria-invalid={false}
        aria-required={false}
        className="MuiInput-input-23 MuiInput-inputType-26"
        disabled={false}
        id="depth"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="number"
        value={28}
      />
    </div>
  </div>
</form>
`;
